bam_qc_metrics/bam_qc.py

```diff
@@ -260,12 +262,17 @@ def _find_metrics(self, input_bam_path):
         self.fast_metrics = self.update_unmapped_count(fast_finder.get_metrics())
         self.logger.info("Finished computing fast bam_qc metrics")
         # apply downsampling (if any)
-        if self.sample_level != None:
+        if self.downsampled_bam != None:
+            self.logger.info("Using downsampled input %s" % self.downsampled_bam)
+            slow_finder_input_path = self.downsampled_bam
+            self.downsampled_total = int(pysam.view('-c', slow_finder_input_path).strip())
+        elif self.sample_level != None:
+            self.logger.info("Downsampling input to %i reads" % self.sample_level)
             total_reads = self.fast_metrics[self.TOTAL_READS_KEY]
             args = [fast_finder_input_path, self.sample_level, total_reads]
```

Comment:
Currently, a supplied `downsampled_bam` does not get quality filtered (MAPQ < 30), making it possible for low quality alignments to be included in slow metrics. I believe quality filtering will also be removed from bam-qc-metrics, so this issue should be temporary.

Reply:
Yes -- the new WDL workflow will filter out undesired reads (non-primary, unmapped, low quality) as well as downsampling.